### PR TITLE
Add jquery.touch w/ git auto-update via single package.json

### PR DESCRIPTION
**Pull request by dakshshah96**
Pull request for issue: #10946
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/ajlkn/jquery.touch
 * Official website (optional, not the repository): 
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required): 
   - Count of watchers: 14
   - Count of stars: 127
   - Count of forks: 36
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/jquery.touch/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "jquery.touch",
+  "filename": "jquery.touch.min.js",
+  "version": "1.1.0",
+  "description": "Adds touch gesture events to jQuery",
+  "keywords": [
+    "jquery",
+    "jquery touch"
```
Comment (extend1994) [inline ajax/libs/jquery.touch/package.json]: Please just use `touch`. We prefer the condition that there is no space between a single keyword expect it has to be more than two words to express specific meaning e.g "Computer Science". Thanks.

**Comment by dakshshah96** [comment]
@extend1994 Done 👍

**Comment by dakshshah96** [comment]
I tried to follow the instructions given by @PeterDaveHello [here](https://www.peterdavehello.org/2015/05/how-to-syncupdate-forked-repository-with-upstream/). However, even after having around 60GB of space on my hard drive, the cloning of the forked repository fails at the `Checking out files..` stage. **Note**: 3.1 GB approximately does get downloaded. The problem is at the later stage.

Any idea what could be the problem?

**Comment by hare1039** [comment]
Hmm, I've no idea on this...
maybe I can rebase this branch. Let me try.

**Comment by hare1039** [comment]
Can I also squash the commits?

**Comment by dakshshah96** [comment]
@hare1039 Sure, go ahead.

**Comment by PeterDaveHello** [comment]
@dakshshah96 you need to enable sparseCheckout

**Comment by hare1039** [comment]
@pvnr0082t Sorry I missed that part. Fix it now.

File: ajax/libs/jquery.touch/package.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  "license": "MIT",
+  "author": "aj <@ajlkn> (https://aj.lkn.io/)",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/ajlkn/jquery.touch.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "jquery.touch*"
```
Comment (clairetsai818) [inline ajax/libs/jquery.touch/package.json]: I think it's better to use `jquery.touch*.(js|map|css)`, so we won't host unrelated files in future updates.

Comment (sufuf3) [inline ajax/libs/jquery.touch/package.json]: `jquery.touch*.(js|map|css)` --> `jquery.touch*.+(js|css|map)`

Comment (dakshshah96) [inline ajax/libs/jquery.touch/package.json]: Which one should I use @clairetsai818 @sufuf3 ?

Comment (sufuf3) [inline ajax/libs/jquery.touch/package.json]: @dakshshah96 please use `jquery.touch*.+(js|css|map)`.
This is a better pattern matcher allowing selection of multiple files ( ref: [node-glob](https://github.com/isaacs/node-glob#glob-primer))
ref: https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md
Thank you.

Comment (clairetsai818) [inline ajax/libs/jquery.touch/package.json]: Oops my fault. Sorry about that.

Comment (PeterDaveHello) [inline ajax/libs/jquery.touch/package.json]: Or maybe just `jquery.touch*` next time since there is no other file types need to be excluded here.

**Comment by PeterDaveHello** [comment]
Thank you all 😘

**Pull request merged**